frontend/src/v5/ui/components/shared/appBar/userMenuButton/userMenuButton.component.tsx

```diff
@@ -0,0 +1,41 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import { MenuIcon, MenuItem, MenuText, Link } from './userMenuButton.styles';
+
+type UserMenuButtonProps = {
+	className?: string;
+	label: string;
+	Icon: any;
+	to?: string;
+	onClickClose: () => void;
+};
+
+export const UserMenuButton = ({ to, Icon, label, onClickClose, ...props }: UserMenuButtonProps) => (
```

Comment:
Maybe we could change this as part of the components for `<DropdownMenu>` and have an icon as optional so we can reuse it in ellipsis menu and in for the avatar menu.
Not sure if the styling is the same though

Reply:
Done. The styling is indeed different. `ellipsis` element take up the whole width (on hover you can see that the background extends to the very boards of the container menu) whereas the `ActionMenu` comes with some padding. However, if we reuse the `ActionMenu` for the `ellipsis` we would have a more consistent UI.